distributed/protocol.py

```diff
@@ -64,35 +63,89 @@
     default_compression = 'lz4'
 
 
+def extract_big_bytes(x):
+    big = {}
+    _extract_big_bytes(x, big)
+    if big:
+        x = deepcopy(x)
+        for path in big:
+            t = get_in(path[:-1], x)
+            if isinstance(t, dict):
+                del t[path[-1]]
+            else:
+                t[path[-1]] = None
+    return x, big
+
+
+def _extract_big_bytes(x, big, path=()):
+    if type(x) is dict:
+        for k, v in x.items():
+            if isinstance(v, (list, dict)):
+                _extract_big_bytes(v, big, path + (k,))
+            elif type(v) is bytes and len(v) >= 2**31:
```

Comment:
This isn't currently the case, but it could be. There are two numbers that are important here:
- The size to pull a bytestring out of the "small" header
- The size to shard each bytestring

Currently these are set to `2**31` and `2**30` respectively. What do you suggest for these numbers instead?